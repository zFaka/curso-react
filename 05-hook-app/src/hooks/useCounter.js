import {useState} from 'react';

const useCounter = (initalState = 0) => {


  const [state, setState] = useState(initalState);


  const increase = (factor = 1) => {

    setState(state + factor);

  }


  const decrease = (factor = 1) => {

    setState(state - factor);

  }


  const reset = () => {

    setState(initalState);

  };



  return {

    state, 
    increase, 
    decrease, 
    reset

  };
}

export default useCounter ;
