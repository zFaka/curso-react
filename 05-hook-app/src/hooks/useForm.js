import {useState} from 'react';

const useForm = (initalState = {})=> {


  const [value, setValue] = useState(initalState);

  const handleInputChange = ({target}) => {

    setValue({
      ...value, 
      [target.name]:target.value
    })
  }


  return [value, handleInputChange];
}

export default useForm;
