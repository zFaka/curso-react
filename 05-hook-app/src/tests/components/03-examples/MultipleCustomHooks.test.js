import {shallow} from "enzyme";
import {MultipleCustomHooks} from "../../../components/03-examples/MultipleCustomHooks";
import useCounter from "../../../hooks/useCounter";
import {useFetch} from "../../../hooks/useFetch";
// Outputs Hooks simulator
jest.mock('../../../hooks/useFetch')
jest.mock("../../../hooks/useCounter")

describe('Test MultipleCustomHooks', () => {



  test('Should show component correctly', () => {


  useCounter.mockReturnValue({

    counter:0, 
    increase:()=>{}
  });
    useFetch.mockReturnValue({data:null,loading:true,error:null});

    const wrapper = shallow(<MultipleCustomHooks/>);
    expect(wrapper).toMatchSnapshot();
  })



  test('Should show info', () => {


  useCounter.mockReturnValue({counter:0, increase:()=>{}});
    useFetch.mockReturnValue({

      data:[{

        author:'Fernando', 
        quote:'Hola Mundo'
      }], 
      loading:false, 
      error:null
    })

    const wrapper = shallow(<MultipleCustomHooks/>);

    expect(wrapper.find('.alert').exists())
      .toBe(false);

    expect(wrapper.find('.mb-3').text().trim())
      .toBe('Hola Mundo');

    expect(wrapper.find('footer').text().trim())
      .toBe('Fernando');
  })

})
