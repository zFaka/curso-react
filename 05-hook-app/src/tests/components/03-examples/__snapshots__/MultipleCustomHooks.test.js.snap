// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MultipleCustomHooks Should show component correctly 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes
  </h1>
  <div
    className="alert alert-info text-center"
  >
    Lo a d  i   n    g
  </div>
</Fragment>
`;
